### ISSUE TITLE
Upgrade to 2.00 cause a lot of pains

### ISSUE THREAD

**Issue by huzhiren** [flutterchina/dio#161]
## Steps to Reproduce

	static Future<dynamic> get(String endpoint, Map<String, dynamic> parameters, { responseType = ResponseType.json }) async {
		print("GET: $endpoint");
		try {
			Map<String, String> headers = {
				"Cookie": "session=${Config.token}",
			};
			Dio dio = Dio();
			final Response response = await dio.get("https://${Config.host}$endpoint", queryParameters: parameters, options: Options(
				headers: headers,
				followRedirects: false,
				responseType: responseType
			));

## Logs

flutter: _url: https://hostname/api/endpoint
flutter: queryParameters: {searchtype: 1, searchinput: , options: 0, sort: U-, search_limit: 10, page: 1}
flutter: DioError [DioErrorType.DEFAULT]: type 'int' is not a subtype of type 'Iterable<dynamic>'#0      _Uri._makeQuery.<anonymous closure> (dart:core/uri.dart:2173:18)
#1      __InternalLinkedHashMap&_HashVMBase&MapMixin&_LinkedHashMapMixin.forEach (dart:collection/runtime/libcompact_hash.dart:367:8)
#2      _Uri._makeQuery (dart:core/uri.dart:2169:21)
#3      new _Uri (dart:core/uri.dart:1512:13)
#4      RequestOptions.uri (package:dio/src/options.dart:197:9)
#5      DefaultHttpClientAdapter.sendRequest (package:dio/src/adapter.dart:86:66)
<asynchronous suspension>

**Comment by pedromorgan** [comment]
v2 does not feel "tested" somehow....

https://github.com/flutterchina/dio/blob/master/lib/src/dio.dart#L44

**Comment by pedromorgan** [comment]
In addition versions are not tagged in git ;-(

**Comment by wendux** [comment]
The `queryParameters` is consistent with `Uri.queryParameters`:
```dart
Map<String, dynamic /*String|Iterable<String>*/ > queryParameters
```
The dynamic type must be one of String or Iterable<String>.

**Comment by wendux** [comment]
@pedromorgan fixed!

**Comment by toanphung** [comment]
In My Project:
`class GetPostParameters {

  int pageIndex;

  int pageSize;

  GetPostParameters({this.pageIndex, this.pageSize});

  Map<String, dynamic> toJson() =>
      {
        'page_index': pageIndex,
        'page_size': pageSize,
      };
}`

When i call api method get with dio.get(url,queryParameters: GetPostParameters.toJson())
I received error:
`I/flutter (24481): DioError [DioErrorType.DEFAULT]: type 'int' is not a subtype of type 'Iterable<dynamic>'#0      _Uri._makeQuery.<anonymous closure> (dart:core/uri.dart:2173:18)
I/flutter (24481): #1      __InternalLinkedHashMap&_HashVMBase&MapMixin&_LinkedHashMapMixin.forEach (dart:collection/runtime/libcompact_hash.dart:367:8)
I/flutter (24481): #2      _Uri._makeQuery (dart:core/uri.dart:2169:21)
I/flutter (24481): #3      new _Uri (dart:core/uri.dart:1512:13)
I/flutter (24481): #4      RequestOptions.uri (package:dio/src/options.dart:201:9)
I/flutter (24481): #5      DefaultHttpClientAdapter.sendRequest (package:dio/src/adapter.dart:78:66)
I/flutter (24481): <asynchronous suspension>
I/flutter (24481): #6      Dio._makeRequest (package:dio/src/dio.dart:653:46)
I/flutter (24481): <asynchronous suspension>
I/flutter (24481): #7      Dio._request.<anonymous closure>.<anonymous closure> (package:dio/src/dio.dart:632:22)
I/flutter (24481): #8      _RootZone.runUnary (dart:async/zone.dart:1379:54)
I/flutter (24481): #9      _FutureListener.handleValue (dart:async/future_impl.dart:126:18)
I/flutter (24481): #10     Future._propagateToListeners.handleValueCallback (dart:async/future_impl.dart:639:45)
I/flutter (24481): #11     Future._propagateToListeners (dart:async/fut`

Before with version dio version 1 it working.

**Comment by huzhiren** [comment]
I think from v2, the value of the parameters should always be converted to string

**Comment by cixiu** [comment]
@toanphung i have same problem

**Comment by wendux** [comment]
fixed! please update to 2.0.5.

**Comment by toanphung** [comment]
It worked. Thanks @wendux
